tests/Unit/Evolution/Systems/GeneralizedHarmonic/GaugeSourceFunctions/Test_DampedHarmonicGaugeQuantities.cpp

```diff
@@ -0,0 +1,1338 @@
+    const tnsr::aa<DataType, SpatialDim, Frame>& pi,
+    const tnsr::iaa<DataType, SpatialDim, Frame>& phi,
+    double exponent) noexcept;
+
+template <size_t SpatialDim, typename Frame, typename DataType>
+void spacetime_deriv_of_power_log_factor(
+    gsl::not_null<tnsr::a<DataType, SpatialDim, Frame>*> d4_powlogfac,
+    const Scalar<DataType>& lapse,
+    const tnsr::I<DataType, SpatialDim, Frame>& shift,
+    const tnsr::A<DataType, SpatialDim, Frame>& spacetime_unit_normal,
+    const tnsr::II<DataType, SpatialDim, Frame>& inverse_spatial_metric,
+    const Scalar<DataType>& sqrt_det_spatial_metric,
+    const tnsr::ii<DataType, SpatialDim, Frame>& dt_spatial_metric,
+    const tnsr::aa<DataType, SpatialDim, Frame>& pi,
+    const tnsr::iaa<DataType, SpatialDim, Frame>& phi, double g_exponent,
+    int exponent) noexcept;
+
+template <size_t SpatialDim, typename Frame, typename DataType>
+tnsr::a<DataType, SpatialDim, Frame> spacetime_deriv_of_power_log_factor(
+    const Scalar<DataType>& lapse,
+    const tnsr::I<DataType, SpatialDim, Frame>& shift,
+    const tnsr::A<DataType, SpatialDim, Frame>& spacetime_unit_normal,
+    const tnsr::II<DataType, SpatialDim, Frame>& inverse_spatial_metric,
+    const Scalar<DataType>& sqrt_det_spatial_metric,
+    const tnsr::ii<DataType, SpatialDim, Frame>& dt_spatial_metric,
+    const tnsr::aa<DataType, SpatialDim, Frame>& pi,
+    const tnsr::iaa<DataType, SpatialDim, Frame>& phi, double g_exponent,
+    int exponent) noexcept;
+}  // namespace DampedHarmonicGauge_detail
+}  // namespace GeneralizedHarmonic
+
+// Wrap `wrap_spacetime_deriv_of_power_log_factor` here to make its last
+// argument a double, allowing for `pypp::check_with_random_values` to work.
+template <size_t SpatialDim, typename Frame, typename DataType>
+tnsr::a<DataType, SpatialDim, Frame> wrap_spacetime_deriv_of_power_log_factor(
+    const Scalar<DataType>& lapse,
+    const tnsr::I<DataType, SpatialDim, Frame>& shift,
+    const tnsr::A<DataType, SpatialDim, Frame>& spacetime_unit_normal,
+    const tnsr::II<DataType, SpatialDim, Frame>& inverse_spatial_metric,
+    const Scalar<DataType>& sqrt_det_spatial_metric,
+    const tnsr::ii<DataType, SpatialDim, Frame>& dt_spatial_metric,
+    const tnsr::aa<DataType, SpatialDim, Frame>& pi,
+    const tnsr::iaa<DataType, SpatialDim, Frame>& phi, const double g_exponent,
+    const double d_exponent) noexcept {
+  const auto exponent = static_cast<int>(d_exponent);
+  tnsr::a<DataType, SpatialDim, Frame> d4_powlogfac{};
+  GeneralizedHarmonic::DampedHarmonicGauge_detail::
+      spacetime_deriv_of_power_log_factor(
+          make_not_null(&d4_powlogfac), lapse, shift, spacetime_unit_normal,
+          inverse_spatial_metric, sqrt_det_spatial_metric, dt_spatial_metric,
+          pi, phi, g_exponent, exponent);
+  return d4_powlogfac;
+}
+
+namespace {
+using Affine = domain::CoordinateMaps::Affine;
+using Affine3D = domain::CoordinateMaps::ProductOf3Maps<Affine, Affine, Affine>;
```

Comment:
You can remove these when you address the comments below

Reply:
Added schwarzschild tests which need these now